### PR TITLE
Add a new github method - GetPullRequestByCommitID

### PR DESCRIPTION
**Pull request by Fredy-Z**
**What this PR does, why we need it**:
1. Add a new github method - `GetPullRequestByCommitID`, so we can find a PR with the commit ID generated from "git squash"
2. `ListPullRequestsWithCommit` is a new method in github client, so I'll need to update the github dependency.

/cc @adrcunha 
/cc @chaodaiG

### CODE REVIEW

File: shared/ghutil/pullrequest.go [hunk 1]
```diff
@@ -133,6 +133,35 @@ func (gc *GithubClient) GetPullRequest(org, repo string, ID int) (*github.PullRe
 	return res, err
 }
 
+// GetPullRequestByCommitID gets PullRequest by commit ID
+func (gc *GithubClient) GetPullRequestByCommitID(org, repo string, commitID string) (*github.PullRequest, error) {
```
Comment (chaodaiG) [inline shared/ghutil/pullrequest.go]: Your may also want to add it to fake client so it's easier for you writing unit tests later

Comment (srinivashegde86) [inline shared/ghutil/pullrequest.go]:
```suggestion
func (gc *GithubClient) GetPullRequestByCommitID(org, repo, commitID string) (*github.PullRequest, error) {
```

File: shared/ghutil/pullrequest.go [hunk 2]
```diff
@@ -133,6 +133,35 @@ func (gc *GithubClient) GetPullRequest(org, repo string, ID int) (*github.PullRe
...
+	var res []*github.PullRequest
+	_, err := gc.retry(
```
Comment (srinivashegde86) [inline shared/ghutil/pullrequest.go]: use combined if?

File: shared/ghutil/pullrequest.go (hunk 1 above)
Comment (Fredy-Z) [reply shared/ghutil/pullrequest.go]: Added.

File: shared/ghutil/pullrequest.go (hunk 2 above)
Comment (Fredy-Z) [reply shared/ghutil/pullrequest.go]: Done. Thanks!

**Comment by chaodaiG** [comment]
/hold
add the method in this interface as well:
https://github.com/knative/test-infra/blob/master/shared/ghutil/client.go

**Comment by Fredy-Z** [comment]
> /hold
> add the method in this interface as well:
> https://github.com/knative/test-infra/blob/master/shared/ghutil/client.go

Added.

**Pull request merged**